### DISCUSSION TITLE
G-code is in inches, but the CNC moves in mm, Intializing GRBL

### DISCUSSION THREAD

**Discussion by dschier** [winder/Universal-G-Code-Sender#2365] [Help]
I got tired of using Win 7 with Mach 3 and a parallel port. With a new PC, GRBL USB interface and UGS I need help. Trying one of my gcode files it looked like it was running in mm, not inches. I have $13 set to 1 and G20 was at the top of the gcode. From info in a closed topic it says to put in a startup line of $N0 = G20. I think if I type that into the command window that UGS will send that to GRBL at every startup. Do I just type that into the command line? Will UGS send it to GRBL each time I connect with UGS? The Mach3 mm/in changover was much different.

**Comment by breiler** [comment]
There are three units in play, the gcode state (G20/G21), the reporting units ($13) and the units displayed in the UGS UI.

To avoid all confusion, UGS will convert the `gcode state units` and the `reporting units` from the controller and display them as the preferred units in the UI. Just click the "change units" icon in the visualizer to toggle between imperial or metric units:
![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/8efc3b62-c430-490f-8e3a-9df16124039c)

There is no need to set the `$N0 = G20` to set the starting units. All that matters is that your gcode file starts with the expected units of that file.

If you run this gcode, it should move one inch. If it doesn't it might mean that your machine isn't calibrated properly and you need check the [GRBL settings](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Configuration#100-101-and-102--xyz-stepsmm).
```
G20
G91
G0 X1
```

**Comment by dschier** [comment]
I think I figured it out. Typing $N0 = G20 into the command line of UGS does fix the problem. Turns out CamBam was inserting G64 at the top of the Gcode on the same line as it put G20 and UGS was ignoring that line.

**Reply by breiler** [reply]
Technically it is GRBL that skips that line. You should probably try and fix those settings in CamBam, the risk is that it is skipping other commands as well.
